From 1cd478f2a4bcd1e4388a6bee8156833b628d78a4 Mon Sep 17 00:00:00 2001
From: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Date: Mon, 19 Nov 2012 10:29:21 +0100
Subject: [PATCH 54/78] wifi: show callstack on sdio access during suspend

BZ: 69319

Preventive patch, with the patch preventing sdio access if
host is suspended, we will loose the kernel stack information
on which function triggered this access. Also reinitialize the
suspended flag in sdio power on (probe function is only
called once at boot).

Change-Id: Ie989ed789c6894c565cded9e28bd37d69605da53
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
Reviewed-on: http://android.intel.com:8080/74594
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Reviewed-by: Casasnovas, QuentinX <quentinx.casasnovas@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/78011
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 wl12xx-compat/drivers/net/wireless/wl12xx/sdio.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/wl12xx-compat/drivers/net/wireless/wl12xx/sdio.c b/wl12xx-compat/drivers/net/wireless/wl12xx/sdio.c
index b8bf9c9..889ada7 100644
--- a/wl12xx-compat/drivers/net/wireless/wl12xx/sdio.c
+++ b/wl12xx-compat/drivers/net/wireless/wl12xx/sdio.c
@@ -78,6 +78,7 @@ static int __must_check wl12xx_sdio_raw_read(struct device *child, int addr,
 
 	if (unlikely(glue->suspended)) {
 		dev_err(child->parent, "prevent sdio read while suspended\n");
+		dump_stack();
 		return -1;
 	}
 
@@ -114,6 +115,7 @@ static int __must_check wl12xx_sdio_raw_write(struct device *child, int addr,
 
 	if (unlikely(glue->suspended)) {
 		dev_err(child->parent, "prevent sdio write while suspended\n");
+		dump_stack();
 		return -1;
 	}
 
@@ -147,6 +149,7 @@ static int wl12xx_sdio_power_on(struct wl12xx_sdio_glue *glue)
 	struct sdio_func *func = dev_to_sdio_func(glue->dev);
 	struct mmc_card *card = func->card;
 
+	glue->suspended = 0;
 	/* Avoid manual resume */
 	func->card->host->bus_resume_flags = 0;
 
-- 
1.7.9

